pkg/common/cliprinter/protojson/protojson.go

```diff
@@ -0,0 +1,83 @@
+package protojson
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/spiffe/spire/pkg/common/cliprinter/errorjson"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+// Print prints one or more protobuf messages formatted as JSON
+func Print(msgs []proto.Message, stdout, stderr io.Writer) {
+	if msgs == nil || len(msgs) == 0 {
+		return
+	}
+
+	jbs := [][]byte{}
+	jm := &protojson.MarshalOptions{
+		UseProtoNames: true,
+	}
+
+	// Unfortunately, we can only marshal one message at a time, so
+	// we need to build up an array of marshaled messages. We do this
+	// before printing them so we don't print an unterminated result
+	for _, msg := range msgs {
+		jb, err := jm.Marshal(msg)
+		if err != nil {
+			errorjson.Print(err, stdout, stderr)
+			return
+		}
+
+		// The golang protobuf encoders state in their documentation
+		// to not rely on the stability of their output. Furthermore,
+		// they deliberately inject randomness into the output to
+		// discourage reliance (and break our tests).
+		//
+		// Since we would like stable output, and we also want parity
```

Comment:
For test stability we may be able to leverage the testify Require/AssertJSONEq methods which IIRC assert semantic equivalence.